todaythediary/src/components/Essentials/Contact.js

```diff
@@ -0,0 +1,193 @@
+import React, { useState, useEffect } from 'react'
+import {
+    Button, Form, FormGroup, Label, Input, Col, FormFeedback
+} from 'reactstrap';
+
+export default function Contact() {
+
+    const [firstName, setFirstName] = useState("");
+    const [lastName, setLastName] = useState("");
+    const [email, setEmail] = useState("");
+    const [telNum, setTelNum] = useState('')
+    const [contactType, setContactType] = useState('Tel.')
+    const [message, setMessage] = useState('')
+    const [active, setActive] = useState(false)
+    const [agree, setAgree] = useState(false)
+    const [touched, setTouched] = useState({
+        firstname: false,
+        lastname: false,
+        telnum: false,
+        email: false
+    })
+    const handleBlur = (field) => (evt) => {
+        //checking whether we visited the fields or not
+        setTouched({
+            ...touched, [field]: true
+        })
+    }
+    function validate(firstName, lastName, telNum, email) {
+
+        //funtion to validate the funtion 
+
+        const errors = {
+            firstName: '',
+            lastName: '',
+            telNum: '',
+            email: '',
+            btn: false
+        };
+
+        if (touched.firstName && firstName.length < 3)
+            errors.firstName = 'First Name should be greater 3 characters';
+        else if (touched.firstName && firstName.length > 10)
+            errors.firstname = 'First Name should be less than 10 characters';
+
+        if (touched.lastName && lastName.length < 3)
+            errors.lastName = 'Last Name should be greater than 3 characters';
+        else if (touched.lastName && lastName.length > 10)
+            errors.lastName = 'Last Name should be less than 10 characters';
+
+        const reg = /^\d+$/;
+        if (touched.telnum && !reg.test(telNum))
+            errors.telNum = 'Tel. Number should contain only numbers ';
+        const validEmailRegex =
+            // eslint-disable-next-line no-useless-escape
+            RegExp(/^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i);
+        if (touched.email) {
+            errors.email =
+                validEmailRegex.test(email)
+                    ? ''
+                    : 'Email is not valid!';
+        }
+        if ((!errors.firstName && !errors.lastName && !errors.telNum && !errors.email)) {
+            errors.btn = true
+        }
+        return errors;
+    }
```

Comment:
It would be good if function `validate` was separated from this component, stored in another file called `helper.js` and imported ashere. This would help you with decreasing the code complexity too.